### ISSUE TITLE
with RSA.importKey on DER format key, "RSA key format is not supported" (w/test case)

### ISSUE THREAD

**Issue by dbachelder** [Legrandin/pycryptodome#188]
We have an app with many pycrypto generated RSA keys in DER format.. We dropped in pycryptodome and have been seeing lots of:

`ValueError: RSA key format is not supported`

when trying to import those keys.

I can't share the keys that are actually failing.. but I created this script to create some fresh ones...

```py
import base64
import sys
from Crypto.PublicKey import RSA
from Cryptodome.PublicKey import RSA as NEW_RSA
successes = 0
fails = []
while successes < 1000:
  k = RSA.generate(2048).exportKey('DER')
  try:
    NEW_RSA.importKey(k)
    sys.stdout.write('.')
    successes += 1
  except ValueError:
    sys.stdout.write('x')
    # just prove this isn't a completely insane key export by importing with pycrypto
    RSA.importKey(k)
    fails.append(base64.urlsafe_b64encode(k))
print '\n'
for f in fails:
  print f
print '\nsuccess %s fails %s' % (successes, len(fails))
```

I get about a 0.8% failure rate... 

here is a more concrete example you don't have to wait for..

```py
import base64
from Cryptodome.PublicKey import RSA as NEW_RSA

NEW_RSA.importKey(base64.urlsafe_b64decode('MIIEpAIBAAKCAQEA6d-pf54zOkvLIcF1Yukmhe1Y2d0f7wg_aev_bHfm2z85yml7XWvtD6eIX5OmoLb_yDRM_1be-guX0tl9SQcJSYYsgFXh8LzxBAgPBL8NCm5BERXNIlbLDo86rPaQfoMTCSJx2hFvD_-ARjNzxuZ6_7FI88HwWibvhhtF5oM-UQ8-5X0mo1Ktgjn4Nj05lZhe-anOzU6z0Brr9WPROyu6gcVo2PBw63uW6-tIg4gvf3DMiEiYjcNbh2IpPi3G5jRJl-CRILGfR1ClrqdPi7JZQM5mhGYf2mLwkcrcf8i5usbH40HDuhoBqmHi5J1ujK8AINyd33fOELHuZDk49Ylq-wIDAQABAoIBAQCqToEzuZQ8Y20sy6AbwwSWilwlam0St8LV81ExP68RAV2oFGANi8Y5nBom33J7nkdtFwjq9-uv0jlhBZfCUillLRvC6Md7c1tfQC8PmcB5VAh11JZo5yM_GvXzdimKtxBpmnemAQQLdleGx3kL4lK3V_Zz60qHm8rYsYuZbQE4LKmArEw7--yh7f_7PD1M65DJ-zGyNbq63cYJ9tT_qgLOlY1SOPTsbodcgrEfxwD3dL276Jy4YRMSMEE5we00bTdDaPKCWOVHPB6pNzzKvLHLXpvw9DkF7zKrHa4NE2AGk-A0r_t_MSrPXfENQOhLrK4UOi-enjqeQdr68hrdVVdBAoGBAO9gKeBSpAQEhiz8BHXXFytg-lepDCwcUYIxp_8wDXH6k8DnvHm9nDaNDPz5eMfL5nyP-Mn3bw4VIlljyLw6oFwcjSnyTsHK5qL59_tw-OGjhlUPqG-rFUAO05KPOi6GxLFITgpOa3hu7rCWZoKBNuTv41gNR0nRwmbQz8_WNsN5AoGBAPodrgKeI-B0At9_VSYjbrJat_FJgwYfPUDInwJ6stCadc2UFxFkT6kXr3e_kpDikHFPadSh-H4Bu-C1EId-JBjSKPOOBZ3HwzenXxzSQdfebAMKGambgitcPBgftgSou7IDYX7vHzzHc3OlQP-Wt81Bc3gydLMKUj7KVcocFPETAoGBALSHn81wRgZJqy-t-WcuCmlKwELwWR_DU62O6ZZJlTpKsrgfjUN7Pp7djdgUBYUZQADdS9J166mBQuCLos5ioKIXXlcuSTtaXMq5Hs7hCrKRczQflPZy3pnrFKFJp6WDOkBBuamILwErGJTROQpfjRjn2luD2CnSKn0EKlYixjZxAoGAO4RSgwb0jJWulVZsga3OypwxxzXx-sP2RHM4-1MNN_x5adkCgD35gshDFeC9VMLFEmGOUm3Ufs-U0IzliQhoOIvR-Nrj1T7S4mxugqI5zs97rmF559U8bWr-VdM98e_g8iRBz_3A1iqz-2HojKllxW78ti5zfKNJHO3y6iwyg4cCgYBqhiV3nHr6oWxhdxUjBkAniY581MrrnMcwVgkj2XIB6-1Vkckruk0XbOfJUJTRTETb6S9sFctyJm5QPVKEjx5WA1jtSw6qIY-9lN9WGkI_brkyo42fo7k9-aQhOkWZMYj2DxS1h641bCGiONg0ET3jMnrzq4gf-jl6sq50wTqBFg=='))
```

I don't suppose this is a known problem with a workaround? It seems like pycrypto asn.1 implementation isn't compatible with pycryptodome?

worth noting that this doesn't seem to be platform dependent.  also 

```sh
python -m Crypto.SelfTest
```

passes 100%

```
pycrypto==2.6
pycryptodomex==3.6.4
```

**Comment by hlawrenz** [comment]
I did a little more digging into this and it appears that the lucas test is incorrectly determining part of the key is not prime. 

The part that is failing is the q factor of the private key. Which in the concrete example above has the value: `175637383534939453397801320455508570374088202376942372758907369518414308188137781042871856139027160010343454418881888953150175357127346872102307696660678617989191485418582475696230580407111841072614783095326672517315988762029036079794994990250662362650625650262324085116467511357592728695033227611029693067539`

I tested this with openssl and it says the number is prime:

```
$ openssl prime 175637383534939453397801320455508570374088202376942372758907369518414308188137781042871856139027160010343454418881888953150175357127346872102307696660678617989191485418582475696230580407111841072614783095326672517315988762029036079794994990250662362650625650262324085116467511357592728695033227611029693067539
FA1DAE029E23E07402DF7F5526236EB25AB7F14983061F3D40C89F027AB2D09A75CD941711644FA917AF77BF9290E290714F69D4A1F87E01BBE0B510877E2418D228F38E059DC7C337A75F1CD241D7DE6C030A19A99B822B5C3C181FB604A8BBB203617EEF1F3CC77373A540FF96B7CD4173783274B30A523ECA55CA1C14F113 is prime
```

**Comment by Woodya** [comment]
fwiw: this fails due to the choice of `10` as the maximal choice of number of `D` "alternates" in `Cryto/Math/Primality.py:lucas_test() .. alternate()`

alternate() produces `[5, -7, 9, -11, 13, -15, 17, -19, 21, -23]` for which the Jacobi(D, n) for @hlawrenz 's prime `n` are all `1`.

the number he found which openssl (1.0.2o) claims is prime requires a `D` of 29 to pass this pre-check.

```py
>>> [Primality.Integer.jacobi_symbol(a, Primality.Integer(n)) for a in [5, -7, 9, -11, 13, -15, 17, -19, 21, -23, 25, -27, 29]].index(-1)
12
```

i'm not sure which code in old `pycrypto` is analogous to this, and is generating pseudo-primes that do not pass the first 10 such possible values of `D`.
it seems that it doesn't use the lucas test, but simply runs the rabin_miller "more" (not sure if enough), after having done a prime sieve using the first 1k primes.

i have verified that if this lucas-test check is caused to pass by checking within the first 13, then the full lucas_test() of this candidate results in `1`.

as written, this is consistent, but it is unable to function when the loaded key is not validated according to this slower/(more-thorough?) test for probable-compositness/probable-primality.

I'm sure that i'm not experienced in this code enough to suggest any particular change, but would it be reasonable to check sufficiently more possible `D`'s when loading/validating a key vs generating fresh keys/checking-primality?
not sure how terrible the code re-org would be vs. generic consequences of looking at arbitrarily more `D` values... or really how many other people have this issue... but @dbachelder 's 0.8% is concerning to us and otherwise upgrading packages seems quite a good idea/important.

of course: the "install both packages and fallback of pycrypto when pycryptodome fails" is a reasonable answer/pov as well. Though we have a library dependency that wants to use the "overwrite" mode, making this at least a little less ideal.

**Comment by dannyroberts** [comment]
@Legrandin thanks for the quick fix for this in https://github.com/Legrandin/pycryptodome/pull/189! Do you have an idea of when a new version (that includes this change) will be available on pypi?

**Comment by Legrandin** [comment]
There should be one today or tomorrow.

**Comment by Legrandin** [comment]
Fixed in v3.6.5
